src/asm/output.c

```diff
@@ -406,11 +396,78 @@ void out_CreatePatch(uint32_t type, struct Expression const *expr)
 		}
 #undef popbyte
 	}
+#undef rpnptr
+}
+
+/*
+ * Allocate a new patch structure and link it into the list
+ */
+static struct Patch *allocpatch(uint32_t type, struct Expression const *expr)
+{
+	struct Patch *pPatch;
+
+	pPatch = malloc(sizeof(struct Patch));
+
+	if (!pPatch)
+		fatalerror("No memory for patch: %s", strerror(errno));
```

Comment:
Don't bother with `strerror()`: it will probably fail, and you know the error anyway. Print it directly and `exit()` immediately.

Reply:
The Linux manual for `strerror` notes that further calls to `strerror` may modify the pointed-to string, implying it's present in a statically-allocated buffer. Therefore, I don't think it would fail.

Reply:
If you're going to assume Linux behavior, then that call is _completely_ pointless, since `malloc()` can only fail with `ENOMEM` under POSIX — meaning that you _know_ the error and you might as well just write it out.

Reply:
`strerror` may localize the error, which may help users not too comfortable with English (and there seem to be plenty), and is more future-proof in case another error code is returned.

Reply:
That was actually a downside in my eyes (I think half-translated messages are awful), but I guess it might be helpful for some people.